rbc/omnisci_backend/__init__.py

```diff
@@ -1,4 +1,5 @@
 from .numpy_funcs import *  # noqa: F401, F403
+from .omnisci_mathimpl import *  # noqa: F401, F403
```

Comment:
IIUC, `omnisci_mathimpl` is not omnisci specific. So, the `omnisci_mathimpl` and `omnisci_mathtyping` should be moved out of `omnisci_backend`. Would it make sense to join these modules into one, say, `rbc/libmath.py` because it mimics what is supported in Python `math` package. 

While supporting Python `math` package from UDF/UDTFs make lots of sense, for the external feature, supporting [cmath](https://en.cppreference.com/w/cpp/header/cmath) makes even more sense as these are the functions that are available when executing the UDF/UDTFs inside the OmniSciDB server.

To make the structure of rbc more clear, I suggest that the external support implementations have dedicated location as follows:
rbc/externals/math.py - provides Python math support as provided in this PR
rbc/externals/cmath.py - provides C math support (not it this PR)
rbc/externals/cstdlb.py - provides C StdLib support ( http://www.cplusplus.com/reference/cstdlib/)
rbc/externals/omniscidb.py - provides OmniSciDB functions declared as `extern "C"` that UDF/UDTFs might need to call.
etc

In this PR, just introduce the `rbc/externals/math.py` module, others will follow in future.

Reply:
> rbc/externals/math.py - provides Python math support as provided in this PR

Notice that this will bind to `math.*` and not `rbc.externals.math.*`.